src/app/code/community/Shopgate/Framework/Model/Payment/Cc/Authncim.php

```diff
@@ -238,15 +238,14 @@ private function _getCardByLastFour($lastFour)
     /**
      * Use this method to check if we can get a $card object
      * Then use regular Authorize when we cannot.
-     * Return true if call is made by redeem_coupons call, as
-     * we just need to validate to get the payment method name.
      *
      * @return bool
      */
     public function checkGenericValid()
     {
-        if ($this->getShopgateOrder() instanceof ShopgateCart) {
-            return true;
+        $cardModel = Mage::getModel('tokenbase/card');
+        if (empty($cardModel)) {
+            return false;
```

Comment:
Would be more convenient to compare $cardModel with false ($cardModel === false), as getModel() will return false or an object instance.